### PR TITLE
Refactor router class

### PR DESCRIPTION
**Pull request by ipabz**

### CODE REVIEW

File: Slim/Router.php [hunk 1]
```diff
@@ -271,12 +272,15 @@ public function getRoutes()
      */
     public function getNamedRoute($name)
     {
-        foreach ($this->routes as $route) {
-            if ($name == $route->getName()) {
-                return $route;
-            }
+        $routes = array_filter($this->routes, function ($route) use ($name) {
```
Comment (geggleto) [inline Slim/Router.php]: This should only ever return one route.

Comment (geggleto) [reply Slim/Router.php]: This is also not an improvement on readability in my opinion.

**Review by geggleto — CHANGES_REQUESTED** [review]

File: Slim/Router.php [hunk 2]
```diff
@@ -233,11 +232,13 @@ protected function createDispatcher()
                 'routeParser' => $this->routeParser,
                 'cacheFile' => $this->cacheFile,
             ]);
-        } else {
-            $this->dispatcher = \FastRoute\simpleDispatcher($routeDefinitionCallback, [
-                'routeParser' => $this->routeParser,
-            ]);
-        }
+
+            return $this->dispatcher;
```
Comment (danopz) [inline Slim/Router.php:236]: Why do you want multiple return statements?

File: Slim/Router.php [hunk 3]
```diff
@@ -272,10 +273,11 @@ public function getRoutes()
     public function getNamedRoute($name)
     {
         foreach ($this->routes as $route) {
-            if ($name == $route->getName()) {
+            if ($name === $route->getName()) {
```
Comment (danopz) [inline Slim/Router.php:276]: Could be a potential BC break?

File: Slim/Router.php (hunk 2 above)
Comment (silentworks) [reply Slim/Router.php:236]: I think its the idea of return early, I am however not sure how much of a speed improvement this gives.

Comment (danopz) [reply Slim/Router.php:236]: Should be nearly nothing as there is no code between the if and return.

File: Slim/Router.php (hunk 3 above)
Comment (geggleto) [reply Slim/Router.php:276]: Yes, it would.

**Comment by silentworks** [comment]
As I don't think there are any benefits to these changes, I will be rejecting this PR.

**Pull request closed without merging**